evol/conditions.py

```diff
@@ -0,0 +1,82 @@
+from time import monotonic
+from typing import Callable, Optional, TYPE_CHECKING
+
+from evol.exceptions import StopEvolution
+
+if TYPE_CHECKING:
+    from evol.population import BasePopulation
+
+
+class Condition:
+    """Stop the evolution until a condition is no longer met.
+
+    :param condition: A function that accepts a Population and returns a boolean.
+        If the condition does not evaluate to True, then the evolution is stopped.
+    """
+    conditions = set()
+
+    def __init__(self, condition: Optional[Callable[['BasePopulation'], bool]]):
+        self.condition = condition
+
+    def __enter__(self):
+        self.conditions.add(self)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
```

Comment:
Why do we need `exc_type`, `exc_val`, `exc_tb` params here? We don't seem to use them.

Reply:
They are required for an `__exit__`